db/db.go

```diff
@@ -92,14 +92,17 @@ type ApplyOpsResponse struct {
 
 // Oplog represents a MongoDB oplog document.
 type Oplog struct {
-	Timestamp primitive.Timestamp `bson:"ts"`
-	HistoryID int64               `bson:"h"`
-	Version   int                 `bson:"v"`
-	Operation string              `bson:"op"`
-	Namespace string              `bson:"ns"`
-	Object    bson.D              `bson:"o"`
-	Query     bson.D              `bson:"o2"`
-	UI        *primitive.Binary   `bson:"ui,omitempty"`
+	Timestamp  primitive.Timestamp `bson:"ts"`
+	HistoryID  int64               `bson:"h"`
+	Version    int                 `bson:"v"`
+	Operation  string              `bson:"op"`
+	Namespace  string              `bson:"ns"`
+	Object     bson.D              `bson:"o"`
+	Query      bson.D              `bson:"o1,omitempty"`
+	UI         *primitive.Binary   `bson:"ui,omitempty"`
+	LSID       bson.Raw            `bson:"lsid,omitempty"`
+	TxnNumber  int64               `bson:"txnNumber,omitempty"`
+	PrevOpTime bson.Raw            `bson:"prevOpTime,omitempty"`
```

Comment:
if I understand correctly, we just parse this field so we can tell whether it was actually set. if it is set, that means this is part of a multi op transaction, and it is not the first entry in the transaction. 
is that true?

Reply:
Close.  It's always present, but for the first (or only) oplog entry of a transaction it is the "empty" value which is defined to mean timestamp of 0,0 and t of -1.